### PR TITLE
Fix Unknown [repository] type [azure] error with 2.2.0

### LINKED ISSUE #16734: Unknown [repository] type [azure] error with Elasticsearch 2.2 & cloud-azure plugin
When using Elasticsearch 2.2.0 and the cloud-azure plugin. When you use the following settings:

https://www.elastic.co/guide/en/elasticsearch/plugins/2.2/cloud-azure-repository.html

```yaml
cloud:
    azure:
        storage:
            my_account:
                account: your_azure_storage_account
                key: your_azure_storage_key
```

When you try to execute the following API you get Unknown [repository] type [azure]:

```sh
[msimos@msi-gs60 elasticsearch-2.2.0]$ curl -XPUT http://localhost:9200/_snapshot/mybackup?pretty -d '{
"type": "azure"
}'
```

```js
{
  "error" : {
    "root_cause" : [ {
      "type" : "repository_exception",
      "reason" : "[mybackup] failed to create repository"
    } ],
    "type" : "repository_exception",
    "reason" : "[mybackup] failed to create repository",
    "caused_by" : {
      "type" : "illegal_argument_exception",
      "reason" : "Unknown [repository] type [azure]"
    }
  },
  "status" : 500
}
```

```sh
[msimos@msi-gs60 elasticsearch-2.2.0]$ curl -XGE http://localhost:9200/_cat/plugins?v
name   component   version type url 
node01 cloud-azure 2.2.0   j        
node01 license     2.2.0   j        
node01 shield      2.2.0   j        
```
In the elasticsearch log file you see this error:

```
[2016-02-19 10:54:47,357][DEBUG][cloud.azure              ] [node01] starting azure services
[2016-02-19 10:54:47,357][DEBUG][cloud.azure              ] [node01] azure repository is not set using [repositories.azure.account] and [cloud.azure.storage.key] properties
```

### PR DESCRIPTION
**Pull request by dadoonet**
This regression has been introduced in 2.2.0 by #13779 where we removed support for `cloud.azure.storage.account` and replaced by `cloud.azure.storage.my_account.account`.
But Azure plugin tries to detect when it starts if all needed settings to use an `azure` repository have been set. If not the case, the plugin does not expose `azure` as an available repository.

Sadly, this check has been badly updated in 2.2.0 so it can never find the expected settings to start correctly.

This gives the following effect:

``` yaml
cloud:
    azure:
        storage:
            my_account:
                account: your_azure_storage_account
                key: your_azure_storage_key
```

When you try to execute the following API you get Unknown [repository] type [azure]:

``` sh
[msimos@msi-gs60 elasticsearch-2.2.0]$ curl -XPUT http://localhost:9200/_snapshot/mybackup?pretty -d '{
"type": "azure"
}'
```

``` js
{
  "error" : {
    "root_cause" : [ {
      "type" : "repository_exception",
      "reason" : "[mybackup] failed to create repository"
    } ],
    "type" : "repository_exception",
    "reason" : "[mybackup] failed to create repository",
    "caused_by" : {
      "type" : "illegal_argument_exception",
      "reason" : "Unknown [repository] type [azure]"
    }
  },
  "status" : 500
}
```

``` sh
[msimos@msi-gs60 elasticsearch-2.2.0]$ curl -XGE http://localhost:9200/_cat/plugins?v
name   component   version type url
node01 cloud-azure 2.2.0   j
node01 license     2.2.0   j
node01 shield      2.2.0   j
```

In the elasticsearch log file you see this error:

```
[2016-02-19 10:54:47,357][DEBUG][cloud.azure              ] [node01] starting azure services
[2016-02-19 10:54:47,357][DEBUG][cloud.azure              ] [node01] azure repository is not set using [repositories.azure.account] and [cloud.azure.storage.key] properties
```

Closes #16734

Note that this does not happen in master branch because we split the azure plugin and we always define `azure` as a repository when the `repository-azure` plugin starts.

### CODE REVIEW

**Comment by dadoonet** [comment]
@imotov Any chance you could look at it?

File: plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java [hunk 1]
```diff
@@ -44,9 +44,6 @@
 
         public static final String TIMEOUT = "cloud.azure.storage.timeout";
 
-        public static final String ACCOUNT = "repositories.azure.account";
-        public static final String LOCATION_MODE = "repositories.azure.location_mode";
-        public static final String KEY = "cloud.azure.storage.key";
         public static final String CONTAINER = "repositories.azure.container";
         public static final String BASE_PATH = "repositories.azure.base_path";
```
Comment (imotov) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java]: It looks like BASE_PATH can be removed as well.

**Comment by imotov** [comment]
Left a minor comment. Otherwise, LGTM.

File: plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java (hunk 1 above)
Comment (dadoonet) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java]: Agreed. I'm still wondering if we should actually support a global `base_path` setting which was the original intention (so I think this is a bug).
Or if we should only support `base_path` for individual repository.

WDYT?

Comment (imotov) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java]: I think we should support it for individual repositories. Not really sure what would be a use case of supporting it globally.

Comment (dadoonet) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageService.java]: It was just for integration tests actually. That way I was able to define this globally and run IT in their own random path name.
Let's remove it than. Thanks!

**Comment by clintongormley** [comment]
@dadoonet is this fix not required in master too?

**Comment by dadoonet** [comment]
No. Master is already working due to some changes we did in master only.

**Pull request merged**